src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.InProcess.cs

```diff
@@ -152,14 +155,31 @@ internal abstract partial class AbstractNavigateToSearchService
         }
 
         private static async Task<ImmutableArray<SearchResult>> ComputeSearchResultsAsync(
-            Project project, Document searchDocument,
+            Project project, ImmutableArray<Document> priorityDocuments, Document searchDocument, 
             PatternMatcher nameMatcher, PatternMatcher containerMatcherOpt,
             DeclaredSymbolInfoKindSet kinds,
             ArrayBuilder<PatternMatch> nameMatches, ArrayBuilder<PatternMatch> containerMatches,
             CancellationToken cancellationToken)
         {
             var result = ArrayBuilder<SearchResult>.GetInstance();
-            foreach (var document in project.Documents)
+
+            // Prioritize the active documents if we have any.
+            var highPriDocs = priorityDocuments.Where(d => project.ContainsDocument(d.Id))
+                                               .ToImmutableArray();
+
+            var highPriDocsSet = highPriDocs.ToSet();
+            var lowPriDocs = project.Documents.Where(d => !highPriDocs.Contains(d));
+
+            var orderedDocs = highPriDocs.AddRange(lowPriDocs);
+
+#if DEBUG
```

Comment:
Why is this necessary?

Reply:
the checks don't have any impact on on the logic of hte fuction.  so i've made them debug-only.